var/spack/repos/builtin/packages/flecsi/package.py

```diff
@@ -57,53 +50,16 @@ class Flecsi(CMakePackage, CudaPackage, ROCmPackage):
     variant("kokkos", default=False, description="Enable Kokkos Support")
     variant("openmp", default=False, description="Enable OpenMP Support")
 
-    # legacy variants
-    variant("coverage", default=False, description="Enable coverage build", when="@:1")
-    variant(
-        "debug_backend", default=False, description="Build Backend with Debug Mode", when="@:1"
-    )
-    variant("disable_metis", default=False, description="Disable FindPackageMetis", when="@:1")
-    variant("doxygen", default=False, description="Enable doxygen", when="@:1")
-    variant("tutorial", default=False, description="Build FleCSI Tutorials", when="@:1")
-    variant("flecstan", default=False, description="Build FleCSI Static Analyzer", when="@:1")
-    variant("external_cinch", default=False, description="Enable External Cinch", when="@:1")
-    variant("unit_tests", default=False, description="Build with Unit Tests Enabled", when="@:1")
-
     # All Current FleCSI Releases
     for level in ("low", "medium", "high"):
-        depends_on("caliper@2.0.1~adiak~libdw", when="@:1 caliper_detail=%s" % level)
-        depends_on("caliper", when="@2.0: caliper_detail=%s" % level)
-        conflicts("^caliper@2.6", when="@2.0: caliper_detail=%s" % level)
-        conflicts("^caliper@2.7", when="@2.0: caliper_detail=%s" % level)
+        depends_on("caliper@2.8:", when=f"@2.0: caliper_detail={level}")
```

Comment:
Does 2.5 not work, or is this just a simplification that shouldn't hurt anyone after so long?

Reply:
The older versions are already marked as deprecated.